src/annotations/components/annotationForm/AnnotationForm.tsx

```diff
@@ -129,11 +145,80 @@ export const AnnotationForm: FC<Props> = (props: Props) => {
     }
   }
 
+  // TODO:  distinguish between editing and creating for this cancel event
   const handleCancel = () => {
     event('dashboards.annotations.create_annotation.cancel')
     props.onClose()
   }
 
+  const isTimeInFuture = timeToCheck => {
+    return moment(timeToCheck).isAfter(moment())
+  }
+
+  const validateStartTime = () => {
+    if (isTimeInFuture(startTime)) {
+      return {isValid: false, message: START_TIME_IN_FUTURE_MESSAGE}
+    }
+    return {isValid: true}
+  }
+
+  /**
+   * If there is a problem with the end time (with respect to the start time)
+   * return an error message here, along with the isValid flag
+   *
+   * If no message, then it is valid (just return a true isValid flag)
+   *
+   * since this is only vaild for range annotations, if it is a point annotations just return true
+   * */
+  const validateEndTime = () => {
+    if (annotationType === 'point') {
+      return {isValid: true}
+    }
+
+    /**
+     * if point annotation, do nothing
+     *
+     * if range:
+     * making sure they are not the same, and also that 'end' is after 'start'
+     */
+
+    const start = moment(startTime)
+    const end = moment(endTime)
+
+    if (end.isSame(start)) {
```

Comment:
another instance to add to the above ticket to handle moment date math https://github.com/influxdata/ui/issues/1844

Reply:
added